test/integration/controller/jobs/job/job_controller_test.go

```diff
@@ -1775,6 +1775,38 @@ var _ = ginkgo.Describe("Job controller interacting with scheduler", ginkgo.Orde
 			wll.Spec.Active = ptr.To(false)
 			gomega.Expect(k8sClient.Update(ctx, wll)).Should(gomega.Succeed())
 
+			gomega.Eventually(func(g gomega.Gomega) {
```

Comment:
Also add similar validations for existing integration tests for PodsReadyTimeout.

I'm interested to see the whole transition from Admitted, to Evicted (Requeued=false), and then Requeue=True.

IIRC, it should be feasible to control the time passing from the test (without using time.Sleep)

Reply:
Added